src/services/nasaService.ts

```diff
@@ -0,0 +1,79 @@
+import axios from 'axios';
+import { Sol } from '../entity/Sol';
+
+interface SolMars {
+  id: string;
+  terrestrial_date: string;
+  sol: string;
+  ls: string;
+  season: string;
+  min_temp: string;
+  max_temp: string;
+  pressure: string;
+  pressure_string: string;
+  abs_humidity: string;
+  wind_speed: string;
+  wind_direction: string;
+  atmo_opacity: string;
+  sunrise: string;
+  sunset: string;
+  local_uv_irradiance_index: string;
+  min_gts_temp: string;
+  max_gts_temp: string;
+}
+
+export class NasaService {
+  private URL: string = 'https://mars.nasa.gov/rss/api/?feed=weather&category=msl&feedtype=json';
+
+  /**
+   *Fetches soles array from the NASA API.
+   * @returns {Promise<SolMars[] | string>} In case of success, returns an array with the data.
+   *   Otherwise, it returns a string with the error message.
+   */
+  private async fetchDataFromNasaApi(): Promise<SolMars[] | string> {
+    try {
+      const response = await axios.get(this.URL);
+      return response.data.soles;
+    } catch (error) {
```

Comment:
faltou tipar o parâmetro de erro

Reply:
Não compreendi. Nesse parâmetro seu eu coloco string por exemplo, como tipo, assim: catch (error: string), recebo o erro: "Catch clause variable type annotation must be 'any' or 'unknown' if specified."